ipapython/kerberos.py

```diff
@@ -181,3 +181,7 @@ def __str__(self):
             principal_string = u'@'.join([principal_string, realm])
 
         return principal_string
+
+    def __repr__(self):
+        return "{}.{}('{}')".format(
+            self.__module__, self.__class__.__name__, self.__str__())
```

Comment:
* does it actually work? Usually instances don't have a ```__module__``` attribute, only classes.
* don't call ```self.__str__()```
* where is the unit test? :)

Something like this should work
```
    def __repr__(self):
        return "{0.__module__}.{0.__name__}('{1}')".format(
            self.__class__, self)
```